### PR TITLE
[tests] Add node_module to turbo cache

### PR DESCRIPTION
**Pull request by styfle**
This is a hack to get turbo to cache `node_modules` so we don't need to rely on the flaky `@actions/cache`.

This reduces the `pnpm install` time from 1 minute down to 15 seconds.

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -2,6 +2,11 @@
   "$schema": "https://turborepo.org/schema.json",
   "globalDependencies": ["turbo-cache-key.json", "test/lib/**"],
   "pipeline": {
+    "//#pnpm-cache-hack": {
```
Comment (Ethan-Arrowood) [inline turbo.json:5]: Very cool - didn't know this syntax existed for running from root

**Review by Ethan-Arrowood — DISMISSED** [review]
LGTM

**Comment by Ethan-Arrowood** [comment]
Oh! Didn't notice you changed to draft - will rereview another time

**Comment by styfle** [comment]
Something isn't being cached/restored correctly because I see a lot of test failures 🤔 

```
src/dev-server.ts(18,25): error TS2307: Cannot find module 'ts-morph' or its corresponding type
```

https://github.com/vercel/vercel/actions/runs/4365824890/jobs/7635101771#step:9:41

**Comment by styfle** [comment]
Closing since this didn't work as expected

**Pull request closed without merging**